### ISSUE TITLE
Segmentation fault in parsing repo XML due to missing error handling around solv_xfopen

### ISSUE THREAD

**Issue by dparalen** [fedora-modularity/fus#50]
In my case, the reproducer is insufficient permissions while reading repo data: 
`(pulp) [vagrant@pulp2 builddir]$ gdb --args ./fus  --repo modulerr,regular,/var/lib/pulp/published/yum/master/yum_distributor/modulerr/1538566893.09/ walrus`
```
# ---->%-----------
Reading symbols from ./fus...done.
(gdb)  run
Starting program: /home/vagrant/devel/fus/builddir/fus --repo modulerr,regular,/var/lib/pulp/published/yum/master/yum_distributor/modulerr/1538566893.09/ walrus
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff6df1868 in fread () at iofread.c:35
35        if (bytes_requested == 0)
Missing separate debuginfos, use: dnf debuginfo-install elfutils-libelf-0.173-1.fc28.x86_64 glib2-2.56.1-4.fc28.x86_64
(gdb) where
#0  0x00007ffff6df1868 in fread () at iofread.c:35
#1  0x00007ffff73d6605 in fread (__stream=0x0, __n=8192, __size=1, __ptr=0x7fffffffb970) at /usr/include/bits/stdio2.h:294
#2  solv_xmlparser_parse (xmlp=xmlp@entry=0x7fffffffd9e0, fp=fp@entry=0x0) at ../ext/solv_xmlparser.c:303
#3  0x00007ffff73c454e in repo_add_repomdxml (repo=<optimized out>, fp=0x0, flags=0) at ../ext/repo_repomdxml.c:355
#4  0x00000000004048a6 in create_repo (pool=0x61b9b0, name=0x63ca70 "modulerr", path=0x63cab0 "/var/lib/pulp/published/yum/master/yum_distributor/modulerr/1538566893.09/") at ../fus.c:416
#5  0x00000000004063da in fus_depsolve (arch=0x7fffffffdd34 "x86_64", platform=0x0, exclude_packages=0x0, repos=0x61b8b0, solvables=0x61b920, error=0x7fffffffddc8) at ../fus.c:985
#6  0x000000000040691d in main (argc=1, argv=0x7fffffffdec8) at ../main.c:69
(gdb) 

```
All works as expected once sufficient permissions are given to the program:
```
(pulp) [vagrant@pulp2 builddir]$ sudo -u apache ./fus --repo modulerr,regular,/var/lib/pulp/published/yum/master/yum_distributor/modulerr/1538566893.09/ walrus
*walrus-5.21-1.noarch@modulerr
module:walrus:5.21:20180704144203:deadbeef.x86_64@modulerr
(null)
(pulp) [vagrant@pulp2 builddir]$ 
```
Having examined the stack, the offending line seems to be the call to [`solv_xfopen()`](https://github.com/fedora-modularity/fus/blob/master/fus.c#L415) that isn't sanitized for return value, which in my case turns out to be `0x0`, causing the `iofread:fread()` to receive a `SIGSEGV`, as can be seen on the stack.

**Comment by igor-raits** [comment]
Yeah, a lot of checks left out of fus due to not having enough time to implement them.

**Comment by r4f4** [comment]
@dparalen Do you think the new checks would help detecting the problem?

**Comment by dparalen** [comment]
@r4f4 thanks for the patch, yeah looks reasonable:

```
(pulp) [vagrant@pulp2 builddir]$  ./fus  --repo f29,source,/var/lib/pulp/published/yum/master/yum_distributor/F29/1539
260489.72/ --repo modulerr,lookaside,/var/lib/pulp/published/yum/master/yum_distributor/modulerr/1538566893.09/ 'modul
e(docker)' --platform f29
Error: Could not open repomd.xml for /var/lib/pulp/published/yum/master/yum_distributor/F29/1539260489.72/: Permission denied
```
As far as the missing files go, what was suggested on the PR makes sense to me; no need to extra check when no data present, the solver failing depsolving is cool here as the input data is invalid/incomplete rather than inaccessible.

**Comment by r4f4** [comment]
Thank you for testing it!

**Comment by dparalen** [comment]
My pleasure! :)

**Comment by r4f4** [comment]
Fixed by ff6ac4da6e1af12764504685cbcfb1083217773d
